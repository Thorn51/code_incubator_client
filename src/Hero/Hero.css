.hero {
  height: 40%;
  display: flex;
  flex-direction: column;
  justify-items: center;
  justify-content: center;
  padding: 30px 0px 30px 0px;
  background: grey;
}

.app_name {
  margin: 40px 0px 0px 0px;
  font-size: 1.5em;
  text-align: center;
}

.hero_tag {
  font-size: 1em;
  padding: 0px 10px 0px 10px;
  text-align: center;
  width: fit-content;
  margin-left: auto;
  margin-right: auto;
}
