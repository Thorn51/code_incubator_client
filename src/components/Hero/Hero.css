.hero {
  height: 40vh;
  display: flex;
  flex-direction: column;
  justify-items: center;
  justify-content: center;
  padding: 30px 0px 30px 0px;
  box-shadow: 0px 0px 15px 0px #002346;
  background: linear-gradient(
        45deg,
        rgba(1, 22, 39, 0.8),
        rgba(0, 35, 170, 0.7),
        rgba(78, 74, 89, 0.7),
        rgba(132, 130, 143, 0.7)
      )
      0 0 / 1000% no-repeat,
    url(https://images.unsplash.com/photo-1569748130764-3fed0c102c59?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80)
      0 0 / cover no-repeat;
  -webkit-animation: gradientAnimation 40s ease infinite;
  animation: gradientAnimation 40s ease infinite;
}
@-webkit-keyframes gradientAnimation {
  0% {
    background-position: 0% 30%, 0 0;
  }
  50% {
    background-position: 100% 70%, 0 0;
  }
  100% {
    background-position: 0% 30%, 0 0;
  }
}
@keyframes gradientAnimation {
  0% {
    background-position: 0% 30%, 0 0;
  }
  50% {
    background-position: 100% 70%, 0 0;
  }
  100% {
    background-position: 0% 30%, 0 0;
  }
}

.app_name {
  margin: 40px 0px 0px 0px;
  color: white;
  font-size: 1.5em;
  text-align: center;
}

.hero_tag {
  font-size: 1em;
  padding: 0px 10px 0px 10px;
  color: white;
  text-align: center;
  width: fit-content;
  margin-left: auto;
  margin-right: auto;
}

@media only screen and (min-width: 600px) {
  .app_name {
    font-size: 2em;
  }
}

@media only screen and (min-width: 1224px) {
  .app_name {
    font-size: 4em;
  }
}
