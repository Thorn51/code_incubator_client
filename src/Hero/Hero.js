import React from "react";
import "./Hero.css";

export default function Hero() {
  return null;
}
