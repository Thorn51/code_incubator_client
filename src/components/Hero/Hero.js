import React from "react";
import "./Hero.css";
import NavBar from "../Navigation/NavBar";

export default function Hero() {
  return (
    <header className="hero">
      <NavBar />
      <h1 className="app_name">Code Incubator</h1>
      <h3 className="hero_tag">
        Grow your web development ideas into full-stack projects
      </h3>
    </header>
  );
}
