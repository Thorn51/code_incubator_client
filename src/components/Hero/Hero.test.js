import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Hero from "./Hero";

describe("<Hero />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Hero />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders HERO as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Hero />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
