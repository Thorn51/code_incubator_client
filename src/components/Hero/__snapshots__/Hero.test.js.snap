// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> Renders HERO as expected 1`] = `
<header
  className="hero"
>
  <h1
    className="app_name"
  >
    Code Incubator
  </h1>
  <h3
    className="hero_tag"
  >
    Grow your web development ideas into full-stack projects
  </h3>
</header>
`;
